### PR TITLE
restart API and added Health into State struct

### PR DESCRIPTION
**Pull request by ragne**
- Added a missed `restart` api
- Added a healthcheck representation to State struct as an Optional (and a test for it)
- Cargo profile for tests; without optimizations should be a bit faster. 
- Added fixtures for a tests instead of using a long unreadable strings. 

I know, that it's probably a lot of changes for a single PR, if you want to review them separately (or if you're not interested in some of them, i.e tests) I'd do so.

### CODE REVIEW

File: src/fixtures/container_inspect_health.json [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+        "Paused": false,
+        "Restarting": false,
+        "OOMKilled": false,
+        "Dead": false,
+        "Pid": 13038,
+        "ExitCode": 0,
+        "Error": "",
+        "StartedAt": "2016-10-25T11:59:38.261828009Z",
+        "FinishedAt": "0001-01-01T00:00:00Z",
+        "Health": {
```
Comment (eldesh) [inline src/fixtures/container_inspect_health.json]: How to get this output?
[The latest API document](https://docs.docker.com/engine/api/v1.39/#operation/ContainerInspect) does not make mention of `"Health"` field.

Comment (ragne) [reply src/fixtures/container_inspect_health.json]: I just dug into docker sources (had a ce17.xx at that time and found that field by experimenting with `inspect` API). It was introduced 3 years ago by this [commit](https://github.com/moby/moby/commit/1a149a0ea59b6653e0ba14599476bfe19c4c33f3) and perhaps wasn't documented?

File: Cargo.toml [hunk 2]
```diff
@@ -55,3 +55,15 @@ unix_socket = "0.5.0"
 
 [target.'cfg(windows)'.dependencies]
 named_pipe = "0.2.2"
+
+
+# The testing profile, used for `cargo test`.
```
[the remark this answers is not in the document]
Comment (ragne) [reply Cargo.toml]: Compiling at opt-level=0 happens faster (at least at my old rig) but often results in a _waaay slower_ code; I saw that at [cargo book](https://doc.rust-lang.org/cargo/reference/manifest.html#the-profile-sections), but I'll happily remove that.

File: src/fixtures/container_inspect_health.json (hunk 1 above)
Comment (ragne) [reply src/fixtures/container_inspect_health.json]: On my current machine 
<details>

Server: Docker Engine - Community                                                                                                                                                                                                                            
Version:          18.09.1   


```json
sudo curl --unix-socket /var/run/docker.sock http://localhost/containers/8f0beab4c79f/json 2>/dev/null | jq -r ".State"                                                                                                           
{                                                                                                                                                                                                                                                            
  "Status": "running",                                                                                                                                                                                                                                       
  "Running": true,                                                                                                                                                                                                                                           
  "Paused": false,                                                                                                                                                                                                                                           
  "Restarting": false,                                                                                                                                                                                                                                       
  "OOMKilled": false,                                                                                                                                                                                                                                        
  "Dead": false,                                                                                                                                                                                                                                             
  "Pid": 14273,                                                                                                                                                                                                                                              
  "ExitCode": 0,                                                                                                                                                                                                                                             
  "Error": "",                                                                                                                                                                                                                                               
  "StartedAt": "2019-02-13T20:20:57.292522202Z",                                                                                                                                                                                                             
  "FinishedAt": "2019-02-13T22:51:00.70155796+03:00",                                                                                                                                                                                                        
  "Health": {                                                                                                                                                                                                                                                
    "Status": "healthy",                                                                                                                                                                                                                                     
    "FailingStreak": 0,                                                                                                                                                                                                                                      
    "Log": [                                                                                                                                                                                                                                                 
      {                                                                                                                                                                                                                                                      
        "Start": "2019-02-13T23:21:27.292902994+03:00",                                                                                                                                                                                                      
        "End": "2019-02-13T23:21:27.565961241+03:00",                                                                                                                                                                                                        
        "ExitCode": 0,                       
        "ExitCode": 0,                                                                                                                                                                                                                                       
        "Output": "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--
     0\r100     2  100     2    0     0     64      0 --:--:-- --:--:-- --:--:--    66\nOK"                                                                                                                                                                  
      },
      ...
  }
}
```
</details>

File: Cargo.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (eldesh) [reply Cargo.toml]: That book says `Listed below is the configuration available, along with the defaults for each profile.`
So, the option `opt-level=0` seems to be set for test profile by default.

File: src/container.rs [hunk 3]
```diff
@@ -181,6 +184,73 @@ pub struct PortMapping {
...
+pub struct LogMessage {
+    pub Start: String,
+    pub End: String,
+    pub ExitCode: u64,
+    pub Output: String,
+}
+
+#[derive(Debug, Clone, Serialize, PartialEq)]
+pub enum HealthState {
+    // Indicates there is no healthcheck
```
Comment (eldesh) [inline src/container.rs]: Please one more `/` for the document if you possibly can 😓

Comment (ragne) [reply src/container.rs]: Sorry, still new to rust :)

**Comment by ragne** [comment]
Thank you for a thorough review!

**Comment by eldesh** [comment]
@ragne Looks good. Lastly, could you rebase to master branch?

**Comment by eldesh** [comment]
:tada:

**Pull request merged**